frontend/app/(tabs)/home.tsx

```diff
@@ -3,23 +3,34 @@ import {
   ScrollView,
   NativeSyntheticEvent,
   NativeScrollEvent,
+  TextInput,
+  TouchableOpacity,
+  Animated,
+  Text,
+  ActivityIndicator,
 } from 'react-native';
-import React, { useState, useCallback, useEffect } from 'react';
+import React, { useState, useCallback, useEffect, useRef } from 'react';
 import { CardData } from '../types/card';
 import { fetchCards, CARDS_PER_PAGE } from '../services/cardService';
 import { Card } from '../components/Card';
 
 const Index = () => {
   const [cards, setCards] = useState<CardData[]>([]);
+  const [filteredCards, setFilteredCards] = useState<CardData[]>([]);
   const [page, setPage] = useState(1);
   const [isLoading, setIsLoading] = useState(false);
+  const [searchText, setSearchText] = useState('');
   const SCROLL_PADDING = 20;
 
+  // Simple animation for search input
+  const searchAnimation = useRef(new Animated.Value(0)).current;
```

Comment:
What is the best practice for animations in react native? Somehow intuitively you would think they are functions but here we create an object and keep a reference around. Is this the usual way to do this?